msal/src/Microsoft.Identity.Client/Internal/PlatformInformationBase.cs

```diff
@@ -34,10 +34,11 @@ namespace Microsoft.Identity.Client.Internal
 {
     internal abstract class PlatformInformationBase : CorePlatformInformationBase
     {
-           
+        public static bool Initialized {get;} = false;
         static PlatformInformationBase()
         {
```

Comment:
Following our current pattern most likely we should ensure ModuleInitizer is called from the PlatformInformationBase static contructor instead of having this two way initializer

Reply:
here the goal of this proposal (which I'm not too happy with) is to force the static constructor of `PlatformInformationBase` to be called (as we know that the static constructor of `ModuleInitializer` is called) so that we are sure that the following instruction get called:

`` Instance = new PlatformInformation(); ``

as currently it can be not called because stuff interact with the base class (CorePlatformInformationBase) and not the derived class? 

I don't understand how ensuring `ModuleInitializer` be called from `PlatformInformationBase` would change the problem?

Reply:
@henrik-me : maybe call PlatformInformationBase static constructor in PlaformPlugIn ?